sno/merge.py

```diff
@@ -148,23 +141,26 @@ def first_true(iterable):
     return next(filter(None, iterable))
 
 
-def handle_merge_conflicts(
-    repo, merge_index, ancestor, ours, theirs,
-):
+def _interactive_pause(prompt):
+    click.prompt(prompt, prompt_suffix="", default="", show_default=False)
+
+
+def resolve_merge_conflicts(repo, merge_index, ancestor, ours, theirs, force=False):
     """
-    Prints merge conflicts and fails, since resolving merge conflicts is not yet supported.
+    Supports resolution of basic merge conflicts, fails in more complex unsupported cases.
 
     repo - a pygit2.Repository
     merge_index - a pygit2.Index containing the attempted merge and merge conflicts.
-    ancestor, ours, theirs - each is either a pygit2.Commit or tuple (pygit2.Commit, label) -
-        where label is a name of the commit that the user would recognize in this context.
-
+    ancestor, ours, theirs - each is a tuple (pygit2.Commit, pygit2.Reference), such
+    as might be obtained from Repository.resolve_refish. The reference can be None.
     """
-    print("Merge conflicts!")
+
+    ANCESTOR, OURS, THEIRS = 0, 1, 2
     # All of the following are 3-tuples of the form (ancestor, ours, theirs)
     commit_args3 = (ancestor, ours, theirs)
-    commits3 = tuple(_get_commit(arg) for arg in commit_args3)
-    labels3 = tuple(_get_commit_label(arg) for arg in commit_args3)
+    commits3 = tuple(arg[0] for arg in commit_args3)
+    refs3 = tuple(arg[1] for arg in commit_args3)
+    labels3 = tuple(_get_long_label(*arg) for arg in commit_args3)
```

Comment:
maybe namedtuple/classify these so it's cleaner to read & understand? ala:

```python
Head = namedtuple('Head', ['commit', 'ref', 'label')  # i guess also needs repo_structure/dataset/pk
Heads = namedtuple('Heads', ['ancestor', 'ours', 'theirs'])

heads = Heads([
  Head(
    commit=h[0],
    label=_get_long_label(*h),
    ref=h[1]
  ) for h in (ancestor, ours, theirs)
])

...

for h in heads:
  print(h.label)

...
heads.ancestor.commit
```

or maybe simpler:
```python
Heads = namedtuple('Heads', ['ancestor', 'ours', 'theirs'])

heads = Heads(ancestor, ours, theirs)
commits = Heads(*(h[0] for h in heads))
refs = Heads(*(h[1] for h in heads))
labels = Heads(*(_get_long_label(*h) for h in heads))
...
labels.ancestor
refs.theirs
```

Reply:
Made a couple of namedtuples -
AncestorOursTheirs - contains 3 versions of basically anything
CommitWithReference - contains (pygit2.Commit, pygit2.Reference)

A bit nicer in places, overall structure is much the same.